addons/mail/static/src/model/model_manager.js

```diff
@@ -384,7 +384,21 @@ export class ModelManager {
     _applyModelDefinition(model) {
         const definition = registry.get(model.name);
         Object.assign(model, Object.fromEntries(definition.get('modelMethods')));
-        Object.assign(model.prototype, Object.fromEntries(definition.get('recordMethods')));
+        const recordMethods = Object.fromEntries(definition.get('recordMethods'));
+        for (let recordMethodName in recordMethods) {
+            if (recordMethodName === 'exists') {
+                // ignore method exists() (otherwise infinite loop)
+                continue;
+            }
```

Comment:
call exists below from model manager instead of from record method so that this is not necessary?